core/src/test/java/org/testcontainers/junit/wait/strategy/HttpWaitStrategyTest.java

```diff
@@ -212,6 +215,30 @@ public void testWaitUntilReadyWithTimoutCausedByReadTimeout() {
         }
     }
 
+    /**
+	 * Test to validate fix from GitHub Pull Request <a href="https://github.com/testcontainers/testcontainers-java/pull/5778">#5778</a>, i.e. when the container startup fails (ContainerLaunchException) before timeout for some reason, we are able to see the root cause of the error in the stack trace, e.g. in this case, a TLS certificate validation error during the TLS handshake test, because we are using a NGINX docker image with self-signed certificate created with the image, that is obviously not trusted.
+	 * The exceptions we should see in the stacktrace ('/' means 'caused by'): ContainerLaunchException / TimeoutException / RuntimeException / SSLHandshakeException / ValidatorException (in sun.* package so not accessible) / SunCertPathBuilderException (in sun.* package so not accessible).
+	 */
+    @Test
+    public void testWaitUntilReadyWithTimeoutCausedBySslHandshakeError() {
+        try (
+            GenericContainer<?> container = new GenericContainer<>(
+                new ImageFromDockerfile()
+                    .withFileFromClasspath("Dockerfile", "https-wait-strategy-dockerfile/Dockerfile")
+                    .withFileFromClasspath("nginx-ssl.conf", "https-wait-strategy-dockerfile/nginx-ssl.conf")
+            )
+                .withExposedPorts(8443)
+                .waitingFor(
+                    createHttpWaitStrategy(ready)
+                        .forPort(8443)
+                        .usingTls()
+                        .withStartupTimeout(Duration.ofMillis(WAIT_TIMEOUT_MILLIS))
+                )
+        ) {
+            waitUntilReadyAndTimeout(container, SSLHandshakeException.class);
```

Comment:
I think it should be enough to use `Assert.assertThrows()` here for the `start()` method, then we can remove all the custom code.

Reply:
If I use `Assert.assertThrows()` on `container.start()`, I only get the first exception in the stacktrace, i.e. `ContainerLaunchException`, I don't get the root cause of this error that I'm looking for, i.e. `SSLHandshakeException`. The purpose of this test is to validate that we are now able to see the root cause of the container launch timeout error in the stacktrace (for troubleshooting). Do you see a way to do this with `Assert.assertThrows()`?

Reply:
```suggestion
            Throwable throwable = Assertions.catchThrowable(container::start);
            assertThat(throwable).hasStackTraceContaining("javax.net.ssl.SSLHandshakeException");
```

Unfortunately, the feature we would ideally need is currently an open PR in AssertJ:
https://github.com/assertj/assertj/pull/2423

However, I think this check would be good enough for our use case and hand and we can remove all the custom code  from `AbstractWaitStrategy`. WDYT?

Reply:
Yes, that's good enough, thanks for the suggestion.